### ISSUE TITLE
automatic imap reconnection issue

### ISSUE THREAD

**Issue by prigaux** [UnivParis1/fcm-login#1]
It correctly detects the disconnection, but fails to reconnect :
```
disconnected, reconnecting...
(node:77483) UnhandledPromiseRejectionWarning: Error: read ECONNRESET
    at TCP.onStreamRead (internal/stream_base_commons.js:111:27)
(node:77483) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:77483) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by prigaux** [comment]
Fixed: f62c742757eea9da714505b55afeb74765667b9a

**Comment by prigaux** [comment]
The commit f62c742757eea9da714505b55afeb74765667b9a "fixes" the issue.
Well it fixes it if you rely on forever/systemd to respawn the process on connection error.
